root/packages/schottky.md

```
@@ -0,0 +1,148 @@
+---
+When implementing LSM-tree or B-tree based key-value stores and database indexes, composite fields often need to be combined into a single byte key. 
+
+Standard serialization formats like JSON or Protocol Buffers are not designed for this purpose because their serialized byte outputs do not preserve the natural sorting order required by unsigned bytewise comparisons (`bytes.Compare` or `memcmp`).
+
+Schottky is a Go library designed to encode multi-type composite tuples into order-preserving byte keys.
+
+```bash
+go get gosuda.org/schottky@latest
+```
+
+### Serialization vs. Sort Keys
```

markdownlint-cli2 (0.23.2)

[warning] Heading levels should only increment by one level at a time
Expected: h2; Actual: h3

(MD001, heading-increment)